nuxeo-runtime/nuxeo-runtime-aws/src/main/java/org/nuxeo/runtime/aws/NuxeoAWSCredentialsProvider.java

```diff
@@ -31,19 +31,49 @@
  */
 public class NuxeoAWSCredentialsProvider implements AWSCredentialsProvider {
 
-    protected static final AWSCredentialsProvider INSTANCE = new NuxeoAWSCredentialsProvider();
-
     protected static final AWSCredentialsProvider DEFAULT = new DefaultAWSCredentialsProviderChain();
 
+    protected final String id;
+
+    /**
+     * Gets a Nuxeo AWS Credentials Provider for the default configuration.
+     */
     public static AWSCredentialsProvider getInstance() {
-        return INSTANCE;
+        return getInstance(null);
+    }
+
+    /**
+     * Gets a Nuxeo AWS Credentials Provider for the given configuration.
+     *
+     * @param id the configuration id, or {@code null} for the default
+     * @since 11.1
+     */
+    public static AWSCredentialsProvider getInstance(String id) {
+        return new NuxeoAWSCredentialsProvider(id);
```

Comment:
It's unfortunate that `getInstance()` creates a new class every time,  that's different from what I expect a "normal" Java   `getInstance()` method to do.  Have you considered caching the instances somehow?  Perhaps with a  `Map<String, AWSCredentialsProvider>`

Reply:
You're right that `getInstance(id)` has become a poor naming. There's no point in caching the instances, these are trivial objects. I think I'll just move callers to use `new NuxeoAWSCredentialsProvider(id)` directly.